### PR TITLE
test: add gateway and organization resource tests

### PR DESCRIPTION
**Pull request by Vaishnav88sk**
test coverage for the gateway and organization management endpoints. It adds unit tests covering `GatewayGroupResource`, `GatewayResource`, and `OrganizationResource`.

Part of #399

### CODE REVIEW

File: control-plane/src/test/java/io/reshapr/ctrl/rest/v1/GatewayGroupResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+/**
+ * @author vaishnav
+ */
+@ExtendWith(MockitoExtension.class)
```
Comment (lbroudoux) [inline control-plane/src/test/java/io/reshapr/ctrl/rest/v1/GatewayGroupResourceTest.java:40]: I think we need to discuss more about which kind of tests we want to focus on.
To be honest - and I apologize, as it may sound rude - but I don't see a lot of value in those purely mocked tests...
* We don't check the service behaviour here
* We don't check the repository behaviour here
* We don't even check the Mappers
* We are just checking that the REST resource is calling either the service, the mappers, or the repository...

Now that we have @QuarkusTests, I think we have a better way (and without much more effort) to check that this chain is actually invoked, with the real component implementations being actually checked and harnessed... 

It may represent way fewer test classes to maintain in the end, with better coverage and confidence that the logic is actually executed.

Could we try this on this single GatewayGroupResource to write this adaptation and see if the benefits are there?

Comment (Vaishnav88sk) [reply control-plane/src/test/java/io/reshapr/ctrl/rest/v1/GatewayGroupResourceTest.java:40]: I completely understand your perspective, and `@QuarkusTest` is fantastic for integration testing and yes, I have thought about this also! However, I'd love to advocate for this alongside integration tests to align with the **industry-standard Testing Pyramid** (popularized by Martin Fowler - I have read some during my college 😄 ). 

few things that were in my mind:

1. **The Testing Pyramid & CI Scalability:** Industry best practice dictate having a wide base of isolated, lightning-fast unit tests, supplemented by a smaller layer of integration tests. Pure Mockito tests run in fractions of milliseconds.
2. **Pinpointing Failures:** If a `@QuarkusTest` fails, the bug could be in the Resource, Service, or DB layer. A mocked unit test isolates the failure to a specific class, drasticaly reducing debugging time..
3. **Tseting Edge Cases:** It lets us easily force the Repository to throw transient exceptions to verify our error-handling logic thoroughly.
4. **Separation of Concerns:** These tests aren't meant to *replace* the integration tests; they supplement them.

But yes, your points are also valid for the repo and do real tests instead of mocks. That was just my thought process. And please do not hesitate to correct me. I have less practical experience, so I may be go in wrong direction and I always accept these as learnings. 😉 So should I go ahead and experiment this `single GatewayGroupResource` as `QuarkusTests`?
